synfig-studio/src/gui/docks/dock_timetrack2.cpp

```diff
@@ -233,6 +233,29 @@ void Dock_Timetrack2::setup_tool_palette()
 		action_button_map[tool_button->get_name()] = tool_button;
 		tool_item_group->add(*tool_button);
 	}
+
+	std::vector<std::string> stock_id{ "synfig-interpolation_type_clamped", "synfig-interpolation_type_tcb",
+								  "synfig-interpolation_type_const", "synfig-interpolation_type_ease" ,
+								  "synfig-interpolation_type_linear"};
+	std::vector<std::string> name{ "Clamped", "TCB" , "Constant" , "Ease In/Out" , "Linear"};
+	std::vector<synfig::Interpolation> interp_type{ synfig::INTERPOLATION_CLAMPED, synfig::INTERPOLATION_TCB,
+												  synfig::INTERPOLATION_CONSTANT, synfig::INTERPOLATION_HALT,
+												  synfig::INTERPOLATION_LINEAR};
+	//button1
+	for (int i=0; i<=4; i++) {

```

Comment:
Prefer [Range-based for loop](https://www.geeksforgeeks.org/range-based-loop-c/).
So it would be good to make a struct with fields: icon_name, name/label, type. Instead of 3 `std::vector`s, we would have one only.
It could be initialized like this:
```c++
std::vector<InterpolationInfo> interpolation_info = {
	{"interpolation_type_clamped_icon", N_("Clamped"), synfig::INTERPOLATION_CLAMPED},
	...
};
```

Reply:
done